observable_operator_test.go

```diff
@@ -647,6 +647,137 @@ func Test_Observable_GroupBy_Error(t *testing.T) {
 	Assert(context.Background(), t, s[2].(Observable), HasAnError())
 }
 
+func Test_Observable_Join(t *testing.T) {
+	type args struct {
+		left, right []interface{}
+		window      Duration
+	}
+	tests := []struct {
+		name     string
+		args     args
+		expected []int64
+	}{
+		{
+			name: "Test 1",
+			args: args{
+				left: []interface{}{
+					map[string]int64{"tt": 1, "V": 1},
+					map[string]int64{"tt": 4, "V": 2},
+					map[string]int64{"tt": 7, "V": 3},
+				},
+				right: []interface{}{
+					map[string]int64{"tt": 2, "V": 5},
+					map[string]int64{"tt": 3, "V": 6},
+					map[string]int64{"tt": 5, "V": 7},
+				},
+				window: WithDuration(2),
+			},
+			expected: []int64{
+				1, 5,
+				1, 6,
+				2, 5,
+				2, 6,
+				2, 7,
+				3, 7,
+			},
+		},
+		{
+			name: "Test 2",
+			args: args{
+				left: []interface{}{
+					map[string]int64{"tt": 1, "V": 1},
+					map[string]int64{"tt": 3, "V": 2},
+					map[string]int64{"tt": 5, "V": 3},
+					map[string]int64{"tt": 9, "V": 4},
+				},
+				right: []interface{}{
+					map[string]int64{"tt": 2, "V": 1},
+					map[string]int64{"tt": 7, "V": 2},
+					map[string]int64{"tt": 10, "V": 3},
+				},
+				window: WithDuration(3),
+			},
+			expected: []int64{
+				1, 1,
+				2, 1,
+				3, 2,
+				4, 2,
+				4, 3,
+			},
+		},
+		{
+			name: "Test with error on left",
+			args: args{
+				left: []interface{}{
+					map[string]int64{"tt": 1, "V": 1},
+					map[string]int64{"tt": 3, "V": 2},
+					errFoo,
+					map[string]int64{"tt": 9, "V": 4},
+				},
+				right: []interface{}{
+					map[string]int64{"tt": 2, "V": 1},
+					map[string]int64{"tt": 7, "V": 2},
+					map[string]int64{"tt": 10, "V": 3},
+				},
+				window: WithDuration(3),
+			},
+			expected: []int64{
+				1, 1,
+				2, 1,
+			},
+		},
+		{
+			name: "Test with error on right",
+			args: args{
+				left: []interface{}{
+					map[string]int64{"tt": 1, "V": 1},
+					map[string]int64{"tt": 3, "V": 2},
+					map[string]int64{"tt": 5, "V": 3},
+					map[string]int64{"tt": 9, "V": 4},
+				},
+				right: []interface{}{
+					map[string]int64{"tt": 2, "V": 1},
+					errFoo,
+					map[string]int64{"tt": 10, "V": 3},
+				},
+				window: WithDuration(3),
+			},
+			expected: []int64{
+				1, 1,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Thanks for your updates.
Yes, we don't use this in RxGo. The key is to be consistent. Either you adapt all the existing tests or you just implement it just like the other tests :wink:

Reply:
Does it look any better ? :)